src/app/beer_garden/db/mongo/pruner.py

```diff
@@ -10,17 +10,30 @@
 
 
 class MongoPruner(StoppableThread):
-    def __init__(self, tasks=None, run_every=None):
+    """
+    Class to prune Mongo DB canceled or timed out Request objects.
+
+    Attributes:
+        logger: Logger to run debug messages
+        display_name: The name of the pruner
+        run_every: How often to run the pruner
+        tasks: List of tasks to prune
+        cancel_threshold: Time after which outstanding requests are marked as canceled
```

Comment:
Should specify the units here (minutes).  Also, the config description mentions that a negative number means never, which is what the behavior should be, but I'm not sure that's properly accounted for in this code.  Setting cancel_threshold to `None` when initializing `MongoPruner` seems like it would just make things blow up and I think you need handling of the negative number option in `app.py` where you're reading the "db.ttl" setting.

It would be good to write an additional unit test for that (the `canceled_threshold=None` / `db.ttl.in_progress < 0` scenario).